docs/examples/multi_gate.rst

```diff
@@ -88,269 +90,509 @@ To run the code in this example, the following is required.
+To start with, solve the circuit with 7 logic gates shown above.
 
-    import dwavebinarycsp
+The code below represents standard Boolean gates with BQMs provided by
+:doc:`dimod </docs_dimod/sdk_index>` BQM generators, represents the Boolean NOT
+operation by inverting the coefficients of the relevant variables, and sums the
+gate-level BQMs. The resulting aggregate BQM has its lowest value for variable
+assignments that satisfy all the constraints representing the circuit's Boolean gates.
 
-    def logic_circuit(a, b, c, d, z):
-        not1 = not b
-        or2 = b or c
-        and3 = a and not1
-        or4 = or2 or d
-        and5 = and3 and or4
-        not6 = not or4
-        or7 = and5 or not6
-        return (z == or7)
-
-    csp = dwavebinarycsp.ConstraintSatisfactionProblem(dwavebinarycsp.BINARY)
-    csp.add_constraint(logic_circuit, ['a', 'b', 'c', 'd', 'z'])
-
-
-2. Multiple small constraints.
-
-.. testcode::
-
-    import dwavebinarycsp
-    import dwavebinarycsp.factories.constraint.gates as gates
-    import operator
-
-    csp = dwavebinarycsp.ConstraintSatisfactionProblem(dwavebinarycsp.BINARY)
-    csp.add_constraint(operator.ne, ['b', 'not1'])  # add NOT 1 gate
-    csp.add_constraint(gates.or_gate(['b', 'c', 'or2']))  # add OR 2 gate
-    csp.add_constraint(gates.and_gate(['a', 'not1', 'and3']))  # add AND 3 gate
-    csp.add_constraint(gates.or_gate(['d', 'or2', 'or4']))  # add OR 4 gate
-    csp.add_constraint(gates.and_gate(['and3', 'or4', 'and5']))  # add AND 5 gate
-    csp.add_constraint(operator.ne, ['or4', 'not6'])  # add NOT 6 gate
-    csp.add_constraint(gates.or_gate(['and5', 'not6', 'z']))  # add OR 7 gate
-
-.. note:: `dwavebinarycsp` works best for constraints of up to 4 variables; it may not
-      function as expected for constraints of over 8 variables.
-
-The next line of code converts the constraints into a BQM that you solve by sampling.
-
-.. testcode::
+>>> from dimod.generators import and_gate, or_gate
+...
+>>> bqm2 = or_gate('b', 'c', 'out2')
+>>> bqm3 = and_gate('a', 'b', 'out3')
+>>> bqm3.flip_variable('b')
+>>> bqm4 = or_gate('out2', 'd', 'out4')
+>>> bqm5 = and_gate('out3', 'out4', 'out5')
+>>> bqm7 = or_gate('out5', 'out4', 'z')
+>>> bqm7.flip_variable('out4')
+...
+>>> bqm = bqm2 + bqm3 + bqm4 + bqm5 + bqm7
+>>> print(bqm.num_variables)
+9
 
-    # Convert the binary constraint satisfaction problem to a binary quadratic model
-    bqm = dwavebinarycsp.stitch(csp)
+This circuit is small enough to solve with a brute-force algorithm such as that
+used by the :class:`~dimod.reference.samplers.ExactSolver` class. Solve and
+print solutions in which the circuit's output, :math:`z`, is 1 (True).
 
-.. toctree::
-   :maxdepth: 1
-   :hidden:
+>>> from dimod import ExactSolver
+...
+>>> solutions = ExactSolver().sample(bqm).lowest()
+>>> z = []
+>>> out_fields = [key for key in list(next(solutions.data(['sample'])))[0].keys() if 'out' in key]
+>>> for datum in solutions.data(['sample', 'energy']):
```

Comment:
Why not just `print(solutions)` and use the pretty printing?

Reply:
The internal variables double the variable count, so it's harder to see ``z=func(a, b, c, d)`` in the output

Reply:
A good future use case for https://github.com/dwavesystems/dimod/pull/806 though IMO

Reply:
Oh, that's a nice PR

Reply:
Reimplemented using https://github.com/dwavesystems/dimod/pull/806 -- anything else needed for approval & merge?